pybossa/sched.py

```diff
@@ -370,42 +370,39 @@ def reserve_task_sql_filters(project_id, reserve_task_keys, exclude):
         return filters, category_keys
 
     # convert task category redis cache key to sql query
-    # ex "co:name:IBM:ticker:IBM_US" would be converted to
-    # "task.info->>'co_name' IN ('IBM')"
-    filters_dict = {}
+    # eg "co_name:IBM:ticker:IBM_US" would be converted to
+    # "task.info->>'co_name' = 'IBM' AND task.info->>'ticker' = 'IBM_US"
+    filter_dict = {}
     current_app.logger.info("Project %s, exclude %s. Build sql filter from reserver task keys", project_id, exclude)
     current_app.logger.info("reserve tasks keys: %s", json.dumps(reserve_task_keys))
     regex_key = "reserve_task:project:{}:category:(.+?):user".format(project_id)
+
     for item in reserve_task_keys:
         data = re.search(regex_key, item)
         if not data:
             continue
 
         category_keys += [item]
         category = data.group(1)
+        if category in filter_dict:
+            continue
```

Comment:
1. How will this work when user can reserve task from same category, but with different values. ex
`company=apple and product=iphone`
`company=samsung and product=galaxy`
it seems continue here will skip second category values
2. How will the final sql be constructed in this case

Reply:
1. IMO the `category` string contains values, therefore the two above category string are different so neither will be skipped, am I missing anything ?

2. final sql is something like:
``` AND ((task.info->>'company' = 'apple' AND task.info->>'product' = 'iphone') OR (task.info->>'company' = 'samsung' AND task.info->>'product' ='galaxy')) IS NOT TRUE```

Reply:
I updated the tests, see [this code ](https://github.com/bloomberg/pybossa/pull/723/files#diff-8091ce91147dac6450bc2f1228f77221d4aacfe988ab7c77bf9db71133b1747aL55-L75)